source/vir/src/ast_to_sst.rs

```diff
@@ -366,6 +366,9 @@ pub(crate) fn expr_to_stm_opt(
             let unique_id = state.get_var_unique_id(&x);
             Ok((vec![], Some(mk_exp(ExpX::Var(unique_id)))))
         }
+        ExprX::VarAt(x, VarAt::Pre) => {
+            Ok((vec![], Some(mk_exp(ExpX::VarAt(x.clone(), VarAt::Pre)))))
```

Comment:
Good point. I don't think a potentially shadowed variable would currently get through the type checker (because one cannot define an `&mut` variable yet), but it's probably good to check nonetheless.

Would the following work, you think? (I don't think I fully understand `rename_map`):
```rust
  ExprX::VarAt(x, VarAt::Pre) => {
      if let Some((scope, _)) = state.rename_map.scope_and_index_of_key(x) {
          if scope != 0 {
              err_str(&expr.span, "the parameter is shadowed here")?;
          }
      }
      Ok((vec![], Some(mk_exp(ExpX::VarAt(x.clone(), VarAt::Pre)))))
  }
```